Cabal/Distribution/Backpack/PreExistingComponent.hs

```diff
@@ -46,9 +48,10 @@ ipiToPreExistingComponent ipi =
     PreExistingComponent {
         pc_pkgname = case Installed.sourcePackageName ipi of
             Just n -> n
-            Nothing -> pkgName $ Installed.sourcePackageId ipi,
+            Nothing -> mkPackageName $ unMungedName
+              $ mungedName $ Installed.sourceMungedId ipi,
```

Comment:
Well hmm, I moved the functionality to `Distribution.InstalledPackageInfo`, and give a it a quick comment, but perhaps the fact that `ghc-pkg` can strip this metadata so care must be taken should also be mentioned.